pulpcore/cli/common.py

```diff
@@ -47,25 +71,24 @@ def wait_for_task(self, task_href, timeout=120):
 
 @click.group()
 @click.version_option(prog_name="pulp3 command line interface")
-@click.option("--base-url", default="https://localhost", help="Api base url")
+@click.option("--config", help="Path to the toml config file")
+@click.option("--base-url", help="Api base url")
 @click.option("--user", default="admin", help="Username on pulp server")
 # @click.option("--password", prompt=True, hide_input=True, help="Password on pulp server")
 @click.option("--password", default="password", help="Password on pulp server")
-@click.option(
-    "--verify-ssl/--no-verify-ssl", default=True, help="Verify SSL connection"
-)
-@click.option(
-    "--format", type=click.Choice(["json"], case_sensitive=False), default="json"
-)
+@click.option("--verify-ssl/--no-verify-ssl", help="Verify SSL connection")
+@click.option("--format", type=click.Choice(["json"], case_sensitive=False))
 @click.pass_context
-def main(ctx, base_url, user, password, verify_ssl, format):
+def main(ctx, config, base_url, user, password, verify_ssl, format):
     ctx.ensure_object(PulpContext)
+
+    ctx.obj.config = PulpConfig(config)
     ctx.obj.api = OpenAPI(
-        base_url=base_url,
+        base_url=(base_url or ctx.obj.config.base_url or "https://localhost"),
```

Comment:
I would prefer to see this logic in the Config object.
If we provide overrides to the __init__ and provide defaults there, too, it's just `ctx.obj.config.base_url` here.